packages/xrpl/test/integration/transactions/delegateSet.test.ts

```diff
@@ -0,0 +1,128 @@
+import { AssertionError } from 'assert'
+
+import { assert } from 'chai'
+
+import {
+  AccountSet,
+  DelegateSet,
+  LedgerEntryResponse,
+  Payment,
+  Wallet,
+  xrpToDrops,
+} from '../../../src'
+import { Delegate } from '../../../src/models/ledger'
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { generateFundedWallet, testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('DelegateSet', function () {
+  let testContext: XrplIntegrationTestContext
+  let alice: Wallet
+  let bob: Wallet
+  let carol: Wallet
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+    alice = testContext.wallet
+    bob = await generateFundedWallet(testContext.client)
+    carol = await generateFundedWallet(testContext.client)
+  })
+  afterEach(async () => teardownClient(testContext))
+
+  it(
+    'no permission',
+    async () => {
+      const tx: Payment = {
+        TransactionType: 'Payment',
+        Account: alice.address,
+        Amount: xrpToDrops(1),
+        Destination: carol.address,
+        Delegate: bob.address,
+      }
+      try {
+        await testTransaction(testContext.client, tx, bob)
+      } catch (err: unknown) {
+        const assertErr = err as AssertionError
+        assert.equal(
+          assertErr.message,
+          "No permission to perform requested operation.: expected 'tecNO_PERMISSION' to equal 'tesSUCCESS'",
+        )
+      }
```

Comment:
Can the framework be adjusted to take other optional error codes instead of needing this?

Reply:
Do you mean adding an optional param like this: `await testTransaction(testContext.client, tx, bob, errCode)`

Reply:
Yes.

Reply:
@mvadari I did the refactor in https://github.com/XRPLF/xrpl.js/pull/2981/commits/760ac8b86cd71372fd9fe256685cc5d720d3fc6d but now only the browser test fails 😅

Reply:
Nevermind, got it working in https://github.com/XRPLF/xrpl.js/pull/2981/commits/2f1f6a264a35e140fdb593283cc83d9f4b1d9215

Reply:
Shouldn't that be used here?

Reply:
Ah yes, missed the update here. Done in https://github.com/XRPLF/xrpl.js/pull/2981/commits/a192433b9b82307230f999dc62620d4c46dbf01e

Reply:
Oh actually, since `tecNO_PERMISSION` is already being tested in `base` flow, I removed it from here.